cmd/cluster/validatepullsecret.go

```diff
@@ -0,0 +1,92 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/openshift/osdctl/pkg/utils"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var BackplaneClusterAdmin = "backplane-cluster-admin"
+
+func newCmdValidatePullSecret(kubeCli client.Client, flags *genericclioptions.ConfigFlags) *cobra.Command {
+	return &cobra.Command{
+		Use:               "validate-pull-secret",
+		Short:             "Checks if the pull secret email matches the owner email",
+		Args:              cobra.ExactArgs(1),
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(ValidatePullSecret(args[0], kubeCli, flags))
+		},
+	}
+}
+
+func ValidatePullSecret(clusterID string, kubeCli client.Client, flags *genericclioptions.ConfigFlags) error {
+	ocm := utils.CreateConnection()
+	defer func() {
+		if ocmCloseErr := ocm.Close(); ocmCloseErr != nil {
+			fmt.Printf("Cannot close the ocm (possible memory leak): %q", ocmCloseErr)
+		}
+	}()
+
+	fmt.Println("Checking if pull secret email matches user email")
+
+	// This is the flagset for the kubeCli object provided from the root command. Set here to retroactively impersonate backplane-cluster-admin
+	flags.Impersonate = &BackplaneClusterAdmin
+	secret := &corev1.Secret{}
+	err := kubeCli.Get(context.TODO(), types.NamespacedName{Namespace: "openshift-config", Name: "pull-secret"}, secret)
+	if err != nil {
+		return err
+	}
+
+	type auth struct {
+		Email string `json:"email"`
+	}
+
+	type auths struct {
+		Cloudopenshiftcom auth `json:"cloud.openshift.com"`
+	}
+
+	type response struct {
+		Auths auths `json:"auths"`
+	}
+
+	dockerConfigJson := response{}
+	err = json.Unmarshal(secret.Data[".dockerconfigjson"], &dockerConfigJson)
```

Comment:
I think we should lookup the value for this key first, to check that it's populated, before attempting to `Unmarshal`:

```
        dockerconfigjsonBytes, found := secret.Data[".dockerconfigjson"]
        if !found {
            // Indicates issue w/ pull-secret, so we can stop evaluating and specify a more direct course of action
            return fmt.Errorf("secret does not contain expected key '.dockerconfigjson'. Do this thing: <link to SOP>")
        }
	err = json.Unmarshal(secret.Data[".dockerconfigjson"], &dockerConfigJson)
```

Reply:
Today I learned! Thanks for sharing this, I totally agree it's a more robust approach.

Reply:
Does anyone have opinions on what next steps should be here? I can't find an sop  that seems relevant for handling this situation. Perhaps one of the following service log templates?

https://github.com/openshift/managed-notifications/blob/master/osd/update_pull_secret.json

https://github.com/openshift/managed-notifications/blob/master/osd/pull_secret_change_breaking_upgradesync.json

Reply:
I've decided to go with the "update_pull_secret" SL. resolving this thread.